### PR TITLE
feat(search): add styling for dark theme - FE-3191

### PR DESCRIPTION
**Pull request by DipperTheDan**
Introduces a new `variant` prop which can be either light or dark. Introduces a new styling for when
search is not focused and search has a value.

fixes FE-3191

### Proposed behaviour

New styling for dark theme

Screenshot 2020-11-27 at 15 36 35

Screenshot 2020-11-27 at 15 36 28

Screenshot 2020-11-27 at 15 36 18

Screenshot 2020-11-27 at 15 36 08

### Current behaviour

Search component does not currently have a `variant` prop or support light or dark themes.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
N/A

### Testing instructions
- `git checkout FE-3191-search_alt_style`
- `npm start` to start Storybook
- Navigate to Design System -> Search -> Search with Alternative Styling
- Navigate to Design System -> Search - > Search with Alternative Styling and No Button
- Designs should match the Figma designs on FE-3191
- Previous styles should not be affected.

### CODE REVIEW

File: src/__experimental__/components/search/search.component.js [hunk 1]
```diff
@@ -117,6 +118,7 @@ const Search = ({
   return (
     <StyledSearch
       searchWidth={searchWidth}
+      searchButton={searchButton}
```
Comment (jamime) [inline src/__experimental__/components/search/search.component.js:121]: Should this be `showSearchButton`?

File: src/__experimental__/components/search/search.style.js [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
+    const darkTheme = variant === "dark";
     return css`
       width: ${searchWidth ? `${searchWidth}` : "100%"};
       padding-bottom: 2px;
       background-color: transparent;
-      border-bottom: 2px solid ${theme.search.passive};
+      border-bottom: 2px solid ${
+        darkTheme
+          ? `${theme.search.darkThemeBorder}`
+          : `${theme.search.passive}`
```
Comment (grabkowski) [inline src/__experimental__/components/search/search.style.js:27]: you could make it a const, e.g.:
```
const variantColor = darkTheme
      ? `${theme.search.darkThemeBorder}`
      : `${theme.search.passive}`
```
and then use it like `border-bottom: ${variantColor};`

File: src/__experimental__/components/search/search.style.js [hunk 3]
```diff
@@ -59,7 +96,27 @@ const StyledSearch = styled.div`
...
+            !searchHasValue &&
+            css`
+              border: 1px solid transparent;
+              color: ${theme.icon.default};
+            `
+          }
+          ${
+            !isFocused &&
+            searchHasValue &&
+            !searchButton &&
```
Comment (grabkowski) [inline src/__experimental__/components/search/search.style.js:110]: this set of conditionals is repeated three times in this file, what do you think of making it a const?

Comment (DipperTheDan) [reply src/__experimental__/components/search/search.style.js:110]: That sounds fine to me. I had thought about making it a `const` but decided against it as the other conditionals were set out this way. I'll get this addressed 👍

File: src/__experimental__/components/search/search.style.js (hunk 2 above)
Comment (DipperTheDan) [reply src/__experimental__/components/search/search.style.js:27]: Yeah good idea. I'll get that done. Thanks.

File: src/__experimental__/components/search/search.component.js (hunk 1 above)
Comment (DipperTheDan) [reply src/__experimental__/components/search/search.component.js:121]: I agree. It's more descriptive more than `searchButton`. I'll get that changed 👍

File: src/__experimental__/components/search/search.style.js [hunk 4]
```diff
@@ -1,17 +1,31 @@
...
+  ${({
+    isFocused,
+    searchWidth,
+    searchIsActive,
+    searchHasValue,
+    searchButton,
+    theme,
+    variant,
+  }) => {
+    const darkTheme = variant === "dark";
```
Comment (jamime) [inline src/__experimental__/components/search/search.style.js:19]: This should probably be `isDarkVariant` too.
```suggestion
    const isDarkVariant = variant === "dark";
```

Comment (DipperTheDan) [reply src/__experimental__/components/search/search.style.js:19]: Yep. Good point. I'll get this changed too.

**Review by grabkowski — DISMISSED** [review]

**Review by jamime — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

File: src/style/themes/base/base-theme.config.js [hunk 5]
```diff
@@ -273,10 +273,15 @@ export default (palette) => {
     },
 
     search: {
-      active: palette.goldTint(50),
+      active: "#FFB500",
```
Comment (jamime) [inline src/style/themes/base/base-theme.config.js:276]: @DipperTheDan can you check with @harpalsingh that we want to change the `border-bottom-color` on `:hover` to be `#FFB500` which is the same as the as the `Textbox` focus outline. Or if it should remain as `goldTint(50)`.

Thanks!

**Review by grabkowski — APPROVED** [review]

**Review by jamime — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by jamime — APPROVED** [review]

**Pull request merged**